spec/decorators/single_spec.rb

```diff
@@ -0,0 +1,52 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+
+require 'spec_helper'
+
+describe ::API::Decorators::Single do
+  let(:user) { FactoryGirl.build(:user, member_in_project: project, member_through_role: role) }
+  let(:project) { FactoryGirl.create(:project_with_types) }
+  let(:role) { FactoryGirl.create(:role, permissions: [:view_work_packages]) }
+  let(:model) { Object.new }
+
+  context 'no user given' do
+    let(:single) { ::API::Decorators::Single.new(model, context: nil) }
+
+    it 'should not authorize an empty user' do
+      expect(single.current_user_allowed_to(:view_work_packages, context: project)).to be_falsey
+    end
+  end
+
+  context 'user given' do
+    let(:single) { ::API::Decorators::Single.new(model, current_user: user) }
+
+    it 'should authorize for a given permission' do
+      expect(single.current_user_allowed_to(:view_work_packages, context: project)).to be_truthy
+    end
```

Comment:
What about a user that does not have a permission?

Wrong implementation that passes all tests in this class:
````
def current_user_allowed_to(*args)
  current_user
end
````